packages/block-library/src/query/index.js

```diff
@@ -22,6 +22,9 @@ export const settings = {
 	edit,
 	save,
 	variations,
+	isMatchingBlockPattern: ( pattern ) => {
+		return pattern.categories?.includes( 'core/query' );
```

Comment:
Is this something we should make implicit? cc @youknowriad 

In the original issue I suggested `patterns: [ ...patternCategories ]`

Reply:
By having a `matching` function will help us fine-tune the behaviour for different use cases. One thing that I'm almost certain will be needed for different contexts (ex. transforms with patterns) will be to also pass the block's attributes.

In addition, if we proceed with this approach and usage of patterns for initializing blocks, we should probably make a convention about the pattern's category name or even consider to augment the block patterns API to have some more properties. Maybe include the block's name somehow? 🤔 

It must be clear for theme/block pattern authors how to make their block patterns shown in this view (block setup)

Reply:
For blocks, we have this `parent` field that allows the author of the block to tell where this block can be inserted. If the idea is similar here, we should try the same approach first. For patterns, it might `context` or `target` or something like that. It all depends on how we want to treat this type of pattern that might not make much sense outside of the given block.

> One thing that I'm almost certain will be needed for different contexts (ex. transforms with patterns) will be to also pass the block's attributes.

We can always add such a function in the future, for now, it feels like a simple match between the pattern and block should be enough.

One idea to consider in the future, similar to how it works with block variations. We could always register patterns together with a block but with PHP:

```
register_block_type_from_metadata( __DIR__, array(
    'patterns' => array(
        $my_pattern, // and so on
    ),
) );

Reply:
>We can always add such a function in the future, for now, it feels like a simple match between the pattern and block should be enough.

I agree 100% :) 

>For blocks, we have this parent field that allows the author of the block to tell where this block can be inserted. If the idea is similar here, we should try the same approach first. For patterns, it might context or target or something like that.

For now, without having it explored in depth yet, the augmentation of `Patterns` API with a new prop makes more sense to me too. One thing that I have in mind is that we will have block patterns that will be used only on block setup, others on transforms etc..

I mention that because there are discussions about transforming a single block to a block pattern which might contain other blocks as well. I think the result would be to replace selected block in pattern and keep the others with their demo content.

This makes it similar with block variations about their `scope` attribute too. I think we could create a new property in `patterns` to include `block name` and such context. An example would be:

```
[ 'core/query-setup', 'core/buttons-setup', 'core/buttons-transforms'
```

I'll explore this a lot better after the designs.